### ISSUE TITLE
clang-tools-extra SerializationTests does not compille on OpenBSD

### ISSUE THREAD

**Issue by brad0** [llvm/llvm-project#53767]
```
/home/brad/llvm-project/clang-tools-extra/clangd/unittests/SerializationTests.cpp:321:20: error: use of undeclared identifier 'RLIMIT_AS'
    if (!getrlimit(RLIMIT_AS, &OriginalLimit)) {
                   ^
/home/brad/llvm-project/clang-tools-extra/clangd/unittests/SerializationTests.cpp:324:30: error: use of undeclared identifier 'RLIMIT_AS'
      Succeeded = !setrlimit(RLIMIT_AS, &NewLimit);
                             ^
/home/brad/llvm-project/clang-tools-extra/clangd/unittests/SerializationTests.cpp:332:17: error: use of undeclared identifier 'RLIMIT_AS'
      setrlimit(RLIMIT_AS, &OriginalLimit);
                ^
3 errors generated.
```

OpenBSD does not have RLIMIT_AS.

Added here.. https://github.com/llvm/llvm-project/commit/3c09103291686630564c1ff3f78c0f8dc69d069f
https://reviews.llvm.org/D91258

**Comment by brad0** [comment]
@sam-mccall ?

**Comment by brad0** [comment]
Posted a diff for review here.. https://reviews.llvm.org/D119989

**Comment by brad0** [comment]
Please merge f374c8ddf2dd4920190cac0ea81e18a74040ddda

[PATCH] [clangd] Fix building SerializationTests unit test on OpenBSD

This fixes building the unit tests on OpenBSD. OpenBSD does not support RLIMIT_AS.

**Comment by tstellar** [comment]
Merged: fef110bf8b2b
